sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,126 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+static const uint32_t kAonTimerFreq = 200 * 1000;  // 200kHz
```

Comment:
Is there a generated version of this?

Reply:
I think it should have, but I didn't manage to find.

Reply:
I think it would be useful to add a generated constant if you want to have a go at doing it.

Reply:
I think you could probably add a `TODO(lowrisc/opentitan#NN): description`, and add the AON Timer clock frequency generation separately. It seems that there's some investigation to be done there.

Reply:
I don't know if it is bug but `aon timer` and `rv timer` are using the same clock frequency (I confurmed with traces) of 125Khz, in this case I can use the constant `kClockFreqPeripheralHz`.

Reply:
I will recommend adding `kClockFreqAonHz` in `sw/device/lib/arch/device.h` and define it to reflect the appropriate value for each of our test running platforms. This is what is done for other clock freqs. This is manually maintained because of testing platform limitations. Kinda surprised that we did not add one for AON clock.

Reply:
I still think it's better to do it as a separate PR, this would make for an easy to review self contained PR.

To begin with we probably should just create:
`const uint64_t kClockFreqAonHz = 200 * 1000;  // 200kHz`

In `device_fpga_cw310.c`, `device_sim_verilator.c`, `device_sim_dv.c`, `device_fpga_nexysvideo.c`.

Reply:
I've created the issue  #9927 to track this.